### PR TITLE
Social Links: pass an argument to get_services() to avoid errors.

### PR DESCRIPTION
**Pull request by jeherve**
#### Changes proposed in this Pull Request:

PHP 7.1 returns a Fatal Error if get_services() is invoked without any argument.
Providing an argument should fix that.

Reported here:
https://wordpress.org/support/topic/publicize-and-wordpress-customizer/

- Related: 3bbf0603784f5eab435316e49cb7d2fb64f88abe where the change was introduced.
- Master issue: #6106

#### Testing instructions:

1. On a server running PHP 7.1, enable a theme using Social Links, like [Ryu](https://wordpress.org/themes/ryu/).
2. Go to Appearance > Customize.
3. Make sure you can set up Social Links without creating any errors.

#### Proposed changelog entry for your changes:
* Social Links: avoid Fatal Errors on sites running PHP 7.1.

### CODE REVIEW

File: modules/theme-tools/social-links.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function customize_register( $wp_customize ) {
 			'priority' => 35,
 		) );
 
-		foreach ( array_keys( $this->publicize->get_services() ) as $service ) {
+		foreach ( array_keys( $this->publicize->get_services( 'connected' ) ) as $service ) {
```
Comment (aduth) [inline modules/theme-tools/social-links.php]: Based on the extended description of 3bbf0603784f5eab435316e49cb7d2fb64f88abe and [fallback behavior of an empty argument](https://github.com/Automattic/jetpack/blob/641796c0585718f2f0725e020ee461a03be3e1e6/modules/publicize/publicize-jetpack.php#L358-L360), I'd expect this should be specified as `'all'`, not `'connected'`.

Comment (jeherve) [inline modules/theme-tools/social-links.php]: Good call! Fixed in cbe29bb82c6c496d59f42222b9cbc90de1258656

**Comment by eliorivero** [comment]
I don't see why we need to pass that argument. It's actually the `Publicize::get_services()` function the one that could be improved by replacing this:
```php
function get_services( $filter ) {
	if ( ! in_array( $filter, array( 'all', 'connected' ) ) ) {
		$filter = 'all';
	}
```
with
```php
function get_services( $filter = 'all' ) {
```
and also update `abstract function get_services( $filter );` to `abstract function get_services( $filter = 'all' );`

In this way, to request all services, we just call `$this->publicize->get_services()` and to get the connected ones, we call `$this->publicize->get_services( 'connected' )`. We save passing the parameter, creating the array, calling `in_array` and so on.

And if this is not clear enough, we can create a couple of intermediate functions `Publicize::get_all_services()` and `Publicize::get_connected_services()` which will in turn call `Publicize::get_services()` with no parameter and with `connected` respectively.

I know I'm probably being a stone in the road to an easy merge, it's just that now that we're fixing this, let's fix it for good. I've created this branch https://github.com/Automattic/jetpack/compare/update/publicize-get_services-signature If you think it's ok, I'll go ahead and close this one and open the other as a proper PR.
Asking @zinigor too for his thoughts.

**Comment by jeherve** [comment]
@eliorivero That makes sense, I agree. I also like the idea of a `get_all_services()` and `get_connected_services()`; it would make it clear what services we're getting, because just calling `get_services()` without any arguments doesn't say much.

**Comment by zinigor** [comment]
Yeah, I'm definitely for two different getter methods, because `get_services( $filter )` feels weird to me. Why not `get_services( $connected_only = true)`?

**Comment by eliorivero** [comment]
Cool, will bring the changes from the other PR into this one.

Yeah, `connected` as boolean works too.

**Pull request merged**